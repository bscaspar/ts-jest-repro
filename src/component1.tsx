import * as React from 'react';
import { View, Text } from 'react-native';
import { Component2 } from '.';

export default () => (
  <View>
    <Component2 />
  </View>
);
