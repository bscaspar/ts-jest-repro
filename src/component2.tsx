import * as React from 'react';
import { View, Text } from 'react-native';

export const Component2 = () => (
  <View>
    <Text>Component2</Text>
  </View>
);
