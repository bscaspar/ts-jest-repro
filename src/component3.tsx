import * as React from 'react';
import { View, Text } from 'react-native';
import { useStore } from './hooks/useStore';

export default () => {
  const context = useStore();
  return (
    <View>
      <Text>Component2</Text>
    </View>
  );
};
