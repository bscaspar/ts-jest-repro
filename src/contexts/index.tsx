import React from 'react';
import { RootStore } from '../../App';

export const RootStoreContext = React.createContext<RootStore | null>(null);
