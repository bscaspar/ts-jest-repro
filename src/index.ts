// export { default as Component1 } from './component1';
// export { default as Component2 } from './component2';
// export { default as Component3 } from './component3'; //commenting this will cause the test to pass
export * from './Component1';
export * from './Component2';
export * from './Component3';
