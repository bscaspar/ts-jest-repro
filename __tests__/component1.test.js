import * as React from 'react';
import { Component1 } from '../src';
import { render } from 'react-native-testing-library';

describe('Component1', () => {
  it('renders correctly', () => {
    const tree = render(<Component1 />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
