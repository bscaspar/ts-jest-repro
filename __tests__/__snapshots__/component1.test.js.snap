// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component1 renders correctly 1`] = `
<View>
  <View>
    <Text>
      Component2
    </Text>
  </View>
</View>
`;
